### DISCUSSION TITLE
Global error handling with fastify adapter

### DISCUSSION THREAD

**Discussion by vovaklimov** [trpc/trpc#2989] [Q&A]
Hello, I have an issue trying to create a global error handler for my tRPC API using fastify adapter. It looks like `onError` handler in `FastifyHandlerOptions` cannot be used, it's return value is ignored, there is no way to change `error` object passed to it and throwing a new error leads to some unexpected results.
So, is there a way I can register a global error handler in order to map errors, thrown by a domain layer, to tRPC ones, to control http status codes, etc?

**Comment by ben-heimberg** [comment]
Did you managed to get it work?
